sklearn/model_selection/_search.py

```diff
@@ -706,9 +708,22 @@ def _store(key_name, array, weights=None, splits=False, rank=False):
                    splits=True, rank=True,
                    weights=test_sample_counts if self.iid else None)
             if self.return_train_score:
+                prev_keys = set(results.keys())
                 _store('train_%s' % scorer_name, train_scores[scorer_name],
                        splits=True)
 
+                if self.return_train_score == 'warn':
+                    for key in set(results.keys()) - prev_keys:
+                        # warn on key access
+                        results.add_warning(
+                            key,
+                            "Computing training scores can slow down cross "
+                            "validation significantly. This is the reason "
+                            "return_train_score will change its default value "
+                            "from True (current behaviour) to False in 0.21. "
+                            "Please set return_train_score explicitly to get "
```

Comment:
I think this message should be changed slightly to say something along the lines of "Looks like you are using training scores, you want to set return_train_score=True because return_train_score default will change in 0.21"